phac-nml/irida-next
# Quality of life changes

## What does this PR do and why?
This PR has a bunch of little changes that improve quality of life
* Add common words to cspell so they stop throwing warnings
* Fix local database not properly initializing when running a local dev environment
* Fix hard crash when running dev environment without saml credentials
* Added a script `/scripts/clean_dev.sh` that can be run after pulling a branch to clean dev environment and prep for testing
  * runs `bundle`
  * drops, rebuilds, and seeds db
* Added a script `/scripts/make_credentials.sh` that can be run to quickly remake credentials file without additional configuration
* Added a docs page for useful dev commands 

## Screenshots or screen recordings
N/A

## How to set up and validate locally
1. Run the script and see that it works
2. using the following vscode configuration, see that it runs without crashing
```
{
    "type": "rdbg",
    "name": "Debug rails",
    "request": "launch",
    "script": "bin/rails s",
    "useBundler": false,
    "command": "ruby",
    "env": {
        "OMNIAUTH_PROVIDERS": "developer,saml,azure_activedirectory_v2"
    }
},
```

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
